src/groups/bmq/bmqc/bmqc_multiqueuethreadpool.h

```diff
@@ -659,37 +556,18 @@ class MultiQueueThreadPool BSLS_KEYWORD_FINAL {
     /// must be passed to one of the `enqueueEvent` methods, or it should be
     /// released with a call to `releaseUnmanagedEvent`, otherwise it will
     /// be leaked.
-    Event* getUnmanagedEvent();
-
-    /// Release the specified `event` which was obtained by a call to
-    /// `getUnmanagedEvent`.
-    void releaseUnmanagedEvent(Event* event);
+    EventSp getEvent();
 
     /// Enqueue the specified `event` on the queue with the specified
     /// `queueId` and return `0` if this `MultiQueueThreadPool` is started,
     /// and return `-1` otherwise'.  Note that if the requested queue is
     /// full, this will block.
-    int enqueueEvent(Event* event, int queueId);
-
-    /// Enqueue the specified `userEvent` on the queue with the specified
-    /// `queueId` and return `0` if this `MultiQueueThreadPool` is started,
-    /// and return `-1` otherwise'.  Note that if the requested queue is
-    /// full, this will block.  WARNING: The behavior is undefined unless
-    /// the specified `userEvent` is a member of an event obtained from a
-    /// call to `getUnmanagedEvent`.
-    int enqueueEvent(TYPE* userEvent, int queueId);
+    int enqueueEvent(bslmf::MovableRef<EventSp> event, int queueId);
 
     /// Enqueue the specified `event` on all queues and return `0` if this
     /// `MultiQueueThreadPool` is started, and return `-1` otherwise'.  Note
     /// that if any queue is full, this will block.
-    int enqueueEventOnAllQueues(Event* event);
-
-    /// Enqueue the specified `event` on all queues and return `0` if this
-    /// `MultiQueueThreadPool` is started, and return `-1` otherwise'.  Note
-    /// that if any queue is full, this will block.  WARNING: The behavior
-    /// is undefined unless the specified `userEvent` is a member of an
-    /// event obtained from a call to `getUnmanagedEvent`.
-    int enqueueEventOnAllQueues(TYPE* userEvent);
+    int enqueueEventOnAllQueues(bslmf::MovableRef<EventSp> event);
```

Comment:
Big fan of taking these raw pointers and turning them into rvalue references 👍🏻.